src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -127,23 +135,33 @@ def get_attrs(
                     ).get("uuid")
             alert_rule_triggers.append(serialized)
 
-        alert_rule_projects = set()
-        for alert_rule in alert_rules.values():
+        alert_activations_ranked = AlertRuleActivations.objects.annotate(
+            rank=Window(
+                expression=RowNumber(),
+                partition_by=[F("alert_rule_id")],
+                order_by=F("date_added").desc(),
```

Comment:
We'll end up with a lot of these per alert rule over time right?

You might need an index on `alert_rule_id, date_added`. I'm not totally sure whether postgres can use an index like this on a window function, so worth validating your query used the appropriate index in redash too.